### PR TITLE
Issue#1 create pages view

### PR DESCRIPTION
**Pull request by jlcollinsilva**
New features for CRUD Pages, including:
- Views and Controllers
- new API for Pages
- Client and Server side validations.

### CODE REVIEW

File: src/NPCMS-Net/API/AccountController.cs [hunk 1]
```diff
@@ -455,6 +455,83 @@ public IActionResult ResetPasswordConfirmation()
                 return View(model);
             }
         }
+        //From here changes related to the project NPCMS 
+
+        //Part of Solution Issue #3 - NPCMS 10/03/2016
+        // GET: /Account/GetAllUsers
+        [HttpGet("GetAllUsers")]
+        [AllowAnonymous]
```
Comment (cdattaway) [inline src/NPCMS-Net/API/AccountController.cs]: I know you have some other changes in here, but I really can't approve a pull request until you get your view models in place and put authentication on these methods.

File: src/NPCMS-Net/API/AccountController.cs [hunk 2]
```diff
@@ -455,6 +455,83 @@ public IActionResult ResetPasswordConfirmation()
...
+
+            return user;
+
+        }
+
+        //Part of Solution Issue #3 - NPCMS 10/04/2016
+        //[HttpDelete("DeleteUser/{useridtodelete}")]
+        [HttpDelete("DeleteUser/{useridtodelete}")]
+        [AllowAnonymous]
+         public async Task<IActionResult> DeleteUser(string useridtodelete)
```
Comment (cdattaway) [inline src/NPCMS-Net/API/AccountController.cs]: Just a reminder to change this to username. We talked about this on Skype.

File: src/NPCMS-Net/API/AccountController.cs [hunk 3]
```diff
@@ -455,6 +455,83 @@ public IActionResult ResetPasswordConfirmation()
...
+        {
+             if (ModelState.IsValid)
+             {
+                 //var user = GetUserById(useridtodelete);
+                 Task<ApplicationUser> user = GetUserById(useridtodelete);
+                if (user == null)
+                 {
+
+                     this.ModelState.AddModelError(null, "User do not exists");
```
Comment (cdattaway) [inline src/NPCMS-Net/API/AccountController.cs]: Grammar: "User does not exist."

File: src/NPCMS-Net/API/NPCMS_PagesController.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        public IEnumerable<NPCMS_Page> Get()
+        {
+            return _npcms_pageservices.ListNPCMS_Pages();
+        }
+
+        // GET api/values/5
+        [HttpGet("{id}")]
+        public NPCMS_Page Get(int id)
+        {
+            return _npcms_pageservices.GetNPCMS_Page(id);
```
Comment (cdattaway) [inline src/NPCMS-Net/API/NPCMS_PagesController.cs]: Use Pascal case or underscore + camel case (ex.: "_camelCase")

Comment (cdattaway) [inline src/NPCMS-Net/API/NPCMS_PagesController.cs]: Also, this applies to the namespace, but I can't comment on that. The Namespace should be NPCMSDotNet or something along that line. It's just naming conventions, so fix it when you can, but it's not extremely important.

File: src/NPCMS-Net/Data/ApplicationDbContext.cs [hunk 5]
```diff
@@ -15,12 +15,16 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
...
+
         protected override void OnModelCreating(ModelBuilder builder)
         {
             base.OnModelCreating(builder);
             // Customize the ASP.NET Identity model and override the defaults if needed.
             // For example, you can rename the ASP.NET Identity table names and more.
             // Add your customizations after calling base.OnModelCreating(builder);
         }
+
+        public DbSet<NPCMS_Page> NPCMS_Pages { get; set; }
```
Comment (cdattaway) [inline src/NPCMS-Net/Data/ApplicationDbContext.cs]: I think you could still call the entity a "Page" however.

**Comment by jlcollinsilva** [comment]
Why? The requirements specify NPCMS_Page name :

All pages should be stored in a database table named NPCMS_Pages.

**Comment by cdattaway** [comment]
I can't figure out how to reply directly to your comment. Anyhow, it's not a big deal, but the naming convention here is grating on me. I suppose if Stephen wants that to be the name of the table, this is fine.

File: src/NPCMS-Net/wwwroot/ngApp/controllers/accountController.ts [hunk 6]
```diff
@@ -52,21 +52,146 @@ namespace NPCMS_Net.Controllers {
...
+                this.clearForm(); 
             }).catch((results) => {
                 this.validationMessages = results;
             });
         }
 
-        constructor(private accountService: NPCMS_Net.Services.AccountService, private $location: ng.ILocationService) { }
+
+        //Part of solution Issue #3 - NPCMS Project 10/03/2016
+        public goAddUser() {
```
Comment (cdattaway) [inline src/NPCMS-Net/wwwroot/ngApp/controllers/accountController.ts]: As in the prior code review, these don't need to exist. a) You should have a single admin page that handles all CRUD functions, and b) use the ui-sref property.

**Comment by jlcollinsilva** [comment]
Done, these changes will be in the next Pull Request, so will have fixes
for issue#1 and issu#3 in one pull request.

File: src/NPCMS-Net/API/AccountController.cs (hunk 1 above)
Comment (jlcollinsilva) [inline src/NPCMS-Net/API/AccountController.cs]: Please verify new Pull request with fixes. Thanks

**Pull request merged**